docs: audit cluster management section (#485)
siderolabs/docs: public/omni/cluster-management/importing-talos-clusters.mdx

description: Import an existing Talos cluster into Omni.
---

> If you have an existing Talos cluster running outside of Omni, you can import it so that Omni can manage it going forward. The import process connects your Talos nodes to Omni, preserves the existing cluster configuration as config patches, and registers the cluster as a managed resource, without resetting or disrupting the running workloads.

Reviewer: ```suggestion
If you have existing Talos clusters running without Omni management, you can import them to be managed by Omni. The import process connects your Talos nodes to Omni, preserves the existing cluster configuration as config patches, and registers the cluster as a managed resource, without resetting or disrupting the running workloads.
```

> > **Note:** This is an experimental feature. Clusters with a custom-built Linux kernel or custom-built extensions are not supported. If your cluster uses either of these, do not proceed with this guide.

Reviewer: I don't think this is considered experimental anymore

  --dry-run
```

A successful dry run will include the following in its output:

```text
cluster "<cluster-name>" is imported successfully but marked as 'locked' to prevent changes done by Omni
 > skipped in dry-run
```

If validation fails, the dry run will report the error and exit without making any changes. Review the error and address any issues before proceeding.

### Step 2: Run the import

Once you are satisfied with the dry run output, run the same command without `--dry-run`:

```shell
omnictl cluster import \
  --talosconfig <path-to-talosconfig> \
  --nodes <node-ip-1>,<node-ip-2>,<node-ip-3> \
  --talos-version <talos-version> \
  --kubernetes-version <kubernetes-version> \
  --initial-talos-version <initial-talos-version> \
  --initial-kubernetes-version <initial-kubernetes-version>
```

A successful import will include the following in its output:

```text
cluster "<cluster-name>" is imported successfully but marked as 'locked' to prevent changes done by Omni
```

The cluster is now registered in Omni but remains locked. No changes will be applied until you explicitly unlock it.

### Step 3: Unlock the cluster

> When you are ready for Omni to begin managing the cluster, unlock it by running the following command, replacing `<cluster-name>` with the name of your cluster:

Reviewer: We should mention what happens when a cluster is unlocked. The endpoint is changed to Omni and other patches will be applied. They can see pending changes in the Omni UI. I think there's a CLI to see the diff too.

This is expected, Omni takes over management of these components as part of the handover.

For more information on cluster locking and unlocking, see [Lock nodes](./upgrading-clusters#lock-nodes).

### Step 4: Rotate CA certificates

All imported clusters are labeled `tainted-by-importing`, which indicates that the cluster secrets are still accessible to anyone with the original `talosconfig` or `kubeconfig`. 

To revoke that access and remove the label, rotate the CA certificates for both the Talos and Kubernetes APIs, replacing `<cluster-name>` with the name of your cluster:

```shell
omnictl cluster -n <cluster-name> secret rotate talos-ca
omnictl cluster -n <cluster-name> secret rotate kubernetes-ca
```

For full details on the rotation process, see [CA Rotation](../cluster-management/ca-rotation).

## How import handles your cluster configuration

Understanding how Omni handles schematics and config patches during import helps you anticipate the outcome and troubleshoot any issues that arise.

> ### Image schematic

Reviewer: This should be higher in the guide and maybe set as a prereq

- Create a matching Schematic in your Omni instance's Image Factory and use it to upgrade the affected nodes before retrying the import.
- Re-run the import with `--force` to skip Schematic validation.

> **Note:** If `--force` is used, Omni will not be able to guarantee upgrade continuity for those nodes using their existing configuration. Omni will still be able to upgrade them, but will create and apply a new Schematic rather than reusing the existing one.

This limitation does not apply to nodes booted directly from [Talos release assets](https://github.com/siderolabs/talos/releases).

### Config patches

To preserve the existing cluster state during import, Omni uses config patches rather than overwriting node configurations.

The import command uses the `--initial-talos-version` and `--initial-kubernetes-version` values to generate the default machine config that Omni would produce for each node. It then compares that default config against the actual config running on each node and generates a config patch representing the difference, effectively capturing all customisations made to the cluster since it was first created.

> Certain machine config fields that are not permitted on Omni-managed clusters are excluded from the generated patches.

Reviewer: We should provide examples

## Abort an import

An in-progress import can be aborted using the following command, replacing `<cluster-name>` with the name of your cluster:

```shell
omnictl cluster import abort <cluster-name>
```

Aborting removes all resources created by Omni during the import and temporarily disconnects Omni from the Talos nodes. Unlike deleting a cluster, aborting an import does not reset the machines the cluster simply stops being managed by Omni.

> **Note:** The cluster must be in `locked` state to abort an import.

During import, Omni applies configuration changes to each node to establish the Omni connection. If the import is aborted, these changes are not automatically reverted. To restore the nodes to their pre-import state, choose the approach that matches your situation:

- **Use Option 1** if the cluster was never unlocked after import and the backed-up configs still reflect the current node state.
- **Use Option 2** if the cluster was unlocked and further modified after import, making the backed-up configs potentially out of date.

> ### Option 1: Restore from backup

Reviewer: Maybe make these tabs
